File: 169-Majority_Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        d = dict()
        for i in nums:
            if i not in d.keys():
                d[i] = 1
            else:
                d[i] += 1
        majority = 0
        for key, val in zip(d.keys(), d.values()):
            if val > floor(len(nums) / 2):
                majority = key
        return majority

File: 169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) { // O(n^2)
        int criteria = (int)(nums.length/2);
        for(int i = 0; i < nums.length; i++){
            int freq = 1;
            for(int j = i+1; j < nums.length; j++){
                if(nums[i] == nums[j])
                    freq++;
            }
            if(freq > criteria)
                return nums[i];
        }
        return 0;
    }
}
